lesson_02_homework.rb

```diff
@@ -1,46 +1,76 @@
 # Code Reading
 
+# We are  defining  the  method  full_name with  3  variables 
 def full_name(first_name, last_name, title)
-  # Example comment line
 
+  # giving name  the  value nil ( no value)
   name = nil
 
-  if title && first_name && last_name
+  # giving if statement  for  the  program. if it  first_name and Last_name  and title  then  it  will run  the  name as  shown  on  the  other  line.
+  if title && first_name && last_name  
     name = title + " " + first_name + " " + last_name
+
+   # If  the  first  condition  is  not  fullfilled because  we  have  title  and  last_name  values then  it  will run  the  second  condition  so  name = title + last_name
   elsif title && last_name
     name = title + " " + last_name
+
+  # if  the  second  condition  is  not completed  it  will  run  the  third  condition  and  name will  be  equal  to  first-name + last_name
   elsif first_name && last_name
     name = first_name + " " + last_name
+   #  if  the  third  condition  is  not  completed  because we  have  just first_name  it  will  run  the  forth  condition  and  name will  be  equal  to  First_name
   elsif first_name
     name = first_name
+    # otherwise  it  will  run  the  last  condition  
   else
     raise "Oh no, that doesn't look like a name"
+    close  the  if  statement
   end
 
+ # It  will  return  the  name defined  by  the  right  condition
   return name
+  #close  the def  method
 end
 
 # Fix Broken Code
 
 def full_name(f, l)
-  puts f + l
+  puts f +" "+ l
 end
 
-puts full_nam("bubba", "chuck")
+puts full_name("bubba", "chuck")
 
 
 # Coding
 
 
 ## Enter your code for your 'add' method below
 
+ def add( first_parameter , second_parameter)
+    sum_1 = first_parameter + second_parameter
+    sum_1
```

Comment:
🔨 @joni88 while this works, this line is redundant. Ruby will automatically return the last line in the method body. so this could be re-written as such:

```
def add(first_number, second_number)
  first_number + second_number
end
```